src/resolvers/reportsInfo.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  Args,
+  ArgsType,
+  Authorized,
+  Ctx,
+  Field,
+  ID,
+  InputType,
+  Int,
+  MiddlewareFn,
+  Mutation,
+  Query,
+  registerEnumType,
+  Resolver,
+  UseMiddleware,
+} from 'type-graphql'
+import { ReportVideoInfo } from '../entities/ReportVideoInfo'
+import { ReportedVideoModel, saveReportedVideo } from '../models/ReportedContent'
+import { OrionContext } from '../types'
+
+const ONE_HOUR = 60 * 60 * 1000
+const MAX_REPORTS_PER_VIDEO = 50
+
+export const rateLimit: (limit: number) => MiddlewareFn<OrionContext> =
+  (limit: number) =>
+  async ({ args, context: { remoteHost } }, next) => {
+    const reportsCountPerVideo = await ReportedVideoModel.count({
+      reporterIp: remoteHost,
+      videoId: args.videoId,
+      timestamp: {
+        $gte: new Date(Date.now() - ONE_HOUR),
+      },
+    })
+
+    if (reportsCountPerVideo > limit) {
+      throw new Error('You have exceeded the maximum number of requests per hour')
+    }
+    return next()
+  }
+
+@ArgsType()
+class ReportVideoArgs {
+  @Field(() => ID)
+  videoId: string
+
+  @Field()
+  rationale: string
+}
+
+enum OrderDirection {
+  ASC = 'asc',
+  DESC = 'desc',
+}
+
+registerEnumType(OrderDirection, { name: 'Order' })
```

Comment:
Could we:
1. Add more specific names as this will be merged with QN schema, to make it clear what this type refers to
2. Use the same sorting convention as QN queries have? So it would be `orderBy: timestamp_ASC`

What do you think?